src/GafferSceneUI/CameraTool.cpp

```diff
@@ -95,6 +95,34 @@ void setValueOrAddKey( Gaffer::FloatPlug *plug, float time, float value )
 	}
 }
 
+/// Assigns a value to something, reassigning the original
+/// value when it goes out of scope. If `experimental::scope_exit`
+/// makes it into the C++ standard, we could use that instead.
+template<typename T>
+class ScopedAssignment : boost::noncopyable
+{
+	public :
+
+		ScopedAssignment( T &target, const T &value )
+			:	m_target( target ), m_originalValue( target )
+		{
+			m_target = value;
+		}
+
+		~ScopedAssignment()
+		{
+			m_target = m_originalValue;
+		}
+
+	private :
+
+		T &m_target;
+		T m_originalValue;
+
+};
+
+bool g_editingTransform = false;
```

Comment:
Side thought whilst rationalising `g_` vs `m_`.  Do we need a more general mechanism to globally block tool/etc updates? Thinking of having two viewers with tools active. If I have my head around it right, then active `TransformTool` tools other than the one being dragged would still update during the drag due to `m_dragging` not `g_dragging`. This might also explain some of the slowness I've seen in my multi-shot experiment.

Reply:
Yep, that makes sense. Not sure what it would look like yet, but definitely something to think about.